src/app/models/Range.js

```diff
@@ -0,0 +1,60 @@
+class Range {
```

Comment:
Want to go through this and how it is implemented in a code walkthrough?

Reply:
Sounds good!